tests/e2e/scale/test_osd_node_balancing.py

```diff
@@ -0,0 +1,187 @@
+"""
+Test osd node balancing by adding nodes and osds and checking their distribution
+"""
+import logging
+from ocs_ci.framework.testlib import scale, ignore_leftovers
+from ocs_ci.ocs import constants, machine, ocp
+from ocs_ci.ocs.node import add_new_node_and_label_it, get_osd_running_nodes, get_nodes
+from ocs_ci.ocs.perftests import PASTest
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.utility.utils import (
+    ceph_health_check,
+    TimeoutSampler,
+    ocsci_log_path,
+)
+
+REPORT = "Node_and_Osd_distribution_report.txt"
+FINAL_REPORT = "Final Report"
+INITIAL_SETUP = "Initial Setup"
+MAX_NODE_COUNT = 9
+MAX_OSDS_PER_NODE = 3
+START_NODE_NUM = 3
+REPLICA_COUNT = 3
+OSD_LIMIT_AT_START = MAX_OSDS_PER_NODE * START_NODE_NUM
+NUM_NODE_ADDS = MAX_NODE_COUNT // REPLICA_COUNT
+
+
+def add_some_worker_nodes():
```

Comment:
Can be just add_worker_nodes and default to 3.

Reply:
changes made